internal/dag/dag.go

```diff
@@ -416,7 +423,28 @@ func (d *DAG) recompute() (dag, IngressrouteStatus) {
 	return _d, IngressrouteStatus{statuses: status, version: d.version}
 }
 
-func (d *DAG) processIngressRoute(ir *ingressroutev1.IngressRoute, prefixMatch string, visited []*ingressroutev1.IngressRoute, host string, service func(m meta, port intstr.IntOrString) *Service, vhost func(host string, port int) *VirtualHost, orphaned map[meta]bool) []Status {
+// returns true if the root ingressroute lives in a root namespace
+func (d *DAG) rootAllowed(ir *ingressroutev1.IngressRoute) bool {
+	if len(d.IngressRouteRootNamespaces) == 0 {
+		return true
+	}
+	for _, ns := range d.IngressRouteRootNamespaces {
+		if ns == ir.Namespace {
+			return true
+		}
+	}
+	return false
```

Comment:
here's a trick; you don't need to do it for this pr

```
for _, ns := range d.IngressRouteRootNamespaces {
		if ns == ir.Namespace {
			return true
		}
	}
	return len(d.IngressRouteRootNamespaces) == 0
```
If `len(d.IngressRouteRootNamespaces)` == 0 then the for loop will execute zero times, and the condition will return true, if there was at least one non matching value in `d.IngressRouteRootNamespaces` then the condition will be false and the function will return false.

Reply:
I think we have this kind of construct in a few places in contour.